src/pages/index.js

```diff
@@ -0,0 +1,273 @@
+    }
+    if (avatar) {
+        profileSelectors.profileAvatar.src = avatar
+    }
+};
+
+api.getUserInfo().then(res => {
+    editUserInfo(res.name, res.about, res.avatar);
+}).catch(err => console.log(err));
+
+const getSubmitButton = (popup) => {
+    return popup.querySelector("button[type='submit']");
+};
+
+const animateText = (button, text) => {
+    let dots = "";
+    return setInterval(() => {
+        dots = dots.length < 3 ? dots + "." : "";
+        button.textContent = `${text}ing${dots}`;
+    }, 500);
+};
+
+const submitButtonText = (popup) => {
+    const submitButton = getSubmitButton(popup);
+    const originalText = submitButton.textContent;
+    return {
+        loadingAnimation: animateText(submitButton, originalText.slice(0, submitButton.textContent.length - 1)),
+        originalText
+    };
+};
+
+const handleEscapeClose = (evt) => {
+    if (evt.key === "Escape") {
+        const openPopup = document.querySelector(".popup.popup_visible");
+        if (openPopup) {
+            togglePopup(openPopup);
+            document.activeElement.blur();
+        }
+    }
+};
+
+const handleMouseClickClose = (evt) => {
+    const openPopup = document.querySelector(".popup.popup_visible");
+    if (openPopup && evt.target === openPopup.children[0]) {
+        togglePopup(openPopup);
+    }
+};
+
+const addPopupListeners = () => {
+    document.addEventListener("keydown", handleEscapeClose);
+    document.addEventListener("click", handleMouseClickClose);
+};
+
+const removePopupListeners = () => {
+    document.removeEventListener("keydown", handleEscapeClose);
+    document.removeEventListener("click", handleMouseClickClose);
+};
+
+function togglePopup(popup) {
+    popup.classList.toggle("popup_visible");
+    const isOpened = popup.classList.contains("popup_visible");
+    isOpened ? addPopupListeners() : removePopupListeners();
+}
+
+function fillEditFormFields() {
+    profileSelectors.editName.value =
+        profileSelectors.profileUserName.textContent.trim();
+    profileSelectors.editDescription.value =
+        profileSelectors.profileDescription.textContent.trim();
+}
+
+function setProfilePopupListeners() {
+    profileSelectors.editButton.addEventListener("click", () => {
+        resetValidation(profileForm);
+        fillEditFormFields();
+        togglePopup(profileSelectors.editPopup);
+    });
+    
+    profileForm.addEventListener("submit", (evt) => {
+        evt.preventDefault();
+        const popup = profileSelectors.editPopup;
+        const { loadingAnimation, originalText } = submitButtonText(popup);
+        api.editUserInfo(profileSelectors.editName.value, profileSelectors.editDescription.value).
+        then(res => editUserInfo(res.name, res.about)).
+        catch(err => console.log(err)).
+        finally(() => {
+            clearInterval(loadingAnimation);
+            getSubmitButton(popup).textContent = originalText;
+            togglePopup(popup);
```

Comment:
`  togglePopup(popup);`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**